### ISSUE TITLE
CursorMoved can be sent after CursorLeft, resulting in invalid cursor positions

### ISSUE THREAD

**Issue by alice-i-cecile** [rust-windowing/winit#3016]
https://github.com/bevyengine/bevy/issues/8840#issuecomment-1594371078

**Comment by kchibisov** [comment]
The issue lacks minimal information to handle it. At least could you provide the platform it was on and winit version you've used. Thx.

**Comment by alice-i-cecile** [comment]
Ping @plonq and @Aceeri for more information.

**Comment by madsmtm** [comment]
And esp. the platform would be useful

**Comment by Plonq** [comment]
I confirm the video is MacOS. Is there any other information I can provide?

**Comment by kchibisov** [comment]
I don't think so, knowing that it was macOS is probably enough.

**Comment by Aceeri** [comment]
I think what we could do is just use the out-of-bounds check PR to also filter events before passing them to users.

**Comment by kchibisov** [comment]
No, that's not how it generally works, but you could use as it as a part of heuristic. Mostly because you have mouse grabs.

**Comment by tomstokes** [comment]
I can replicate a similar issue with CursorMoved events firing before CursorEntered on macOS (15.4.1). Tested with winnit 0.30.10.

The most reliable way I've found to replicate:
1. Move window using the title bar
2. Release mouse click
3. Move pointer downward into window frame

This creates a CursorMoved event with a very small y value, followed by a CursorEntered event.

Bringing the cursor in from other edges can sometimes produce the behavior, but the sequence above has been most reliable.
